cmake/SpectreSetupPythonPackage.cmake

```diff
@@ -33,6 +33,10 @@ configure_or_symlink_py_file(
 )
 # Also link the main entry point to bin/
 set(PYTHON_SCRIPT_LOCATION "spectre")
+set(JEMALLOC_PRELOAD "")
+if(BUILD_PYTHON_BINDINGS AND "${JEMALLOC_LIB_TYPE}" STREQUAL SHARED)
+  set(JEMALLOC_PRELOAD "LD_PRELOAD=${JEMALLOC_LIBRARIES}")
```

Comment:
Not for this PR, but my general education... @wthrowe would there be issues here if someone did `LD_PRELOAD=... ./bin/specter ...` where either our or the user's `LD_PRELOAD` is overwritten/ignored?

Reply:
This will overwrite any preexisting value, so if anyone is relying on that it could cause problems.  Properly adding to these colon-separated values is annoying.  I think the most correct thing would be `LD_PRELOAD=${LD_PRELOAD}${LD_PRELOAD:+:}new_lib`.  That middle thing can be read as "Look at `LD_PRELOAD`, and, treating unset as empty (`:`), if it is not empty then substitute (`+`) the string `:`."

I'm not sure whether prepending or appending is correct here.  The above appends, prepending can be done by reversing the order of the three sections.

Reply:
squashed a change to append if LD_PRELOAD is defined